cloudinit/util.py

```diff
@@ -72,6 +72,8 @@
 
 _LSB_RELEASE = {}
```

Comment:
Maybe it's worth creating a cache dict that multiple functions can use:
```suggestion
_CACHED_RESPONSES = {}
```
Maybe it's worth creating a cache dict that multiple functions can use.
Then you could adapt lsb_release and is_FreeBSD to something like the following:

```
global _CACHED_RESPONSES

if 'lsb_release' in _CACHED_RESPONSE:
    data = _CACHED_RESPONSE.get('lsb_release')
else:
   data = _lsb_release()
return data
```

Reply:
if only there was a `@memoize` decorator …

Reply:
given discussions on IRC, I think it would, indeed, be smart to create a custom decorator, and memoize as much as possible, especially here in utils